node_modules/gh-auth/lib/oauth/rest.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2015 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+
+var GrassHopper = require('gh-core');
+var GrassHopperUtil = require('gh-core/lib/util');
+
+var AuthOauth = require('../oauth');
+var AuthOauthServer = require('./server');
+
+/**
+ * Perform the OAuth 2.0 authorization code grant operation
+ *
+ * @param  {Request}    req     The request object performing the grant
+ * @param  {Response}   res     The response object on which to send the response
+ * @api private
+ */
+var _grant = function(req, res) {
+    // Only "code" authorization workflow is supported
+    var responseType = req.body.response_type;
+    if (responseType !== 'code') {
+        return GrassHopperUtil.sendError(res, {'code': 401, 'msg': 'Only "code" is supported for response type'});
+    }
+
+    // Note that the grant flow also has a redirect_uri, but we don't acknowledge it here. Since we
+    // currently don't support multiple redirect URIs, it's not necessary and instead always
+    // redirect to the configured redirect URI for simplicity
+
+    // Perform the validation, authorization and grant operation
+    AuthOauth.grantAuthCode(req.ctx, req.body.client_id, req.body.for_user_id, function(err, redirectUri) {
+        if (err) {
+            return GrassHopperUtil.sendError(res, err);
+        }
+
+        // Redirect to the specified redirect uri
+        return res.redirect(redirectUri);
```

Comment:
This redirect might cause some issues if we're going to build a UI which does the grant + getAccessToken steps.
IIRC, you can't perform an AJAX call and NOT follow redirects.